public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -600,58 +600,83 @@ export class ElasticDatasource
     return [SupplementaryQueryType.LogsVolume];
   }
 
-  getLogsVolumeDataProvider(request: DataQueryRequest<ElasticsearchQuery>): Observable<DataQueryResponse> | undefined {
-    const isLogsVolumeAvailable = request.targets.some((target) => {
-      return target.metrics?.length === 1 && target.metrics[0].type === 'logs';
-    });
-    if (!isLogsVolumeAvailable) {
+  getSupplementaryQuery(type: SupplementaryQueryType, query: ElasticsearchQuery): ElasticsearchQuery | undefined {
+    if (!this.getSupportedSupplementaryQueryTypes().includes(type)) {
       return undefined;
     }
+
+    let isQuerySuitable = false;
+    let supplementaryQuery: ElasticsearchQuery | undefined = undefined;
+
+    switch (type) {
+      case SupplementaryQueryType.LogsVolume:
+        // it has to be a logs-producing range-query
+        isQuerySuitable = !!(query.metrics?.length === 1 && query.metrics[0].type === 'logs');
+        if (isQuerySuitable) {
```

Comment:
I would suggest to try to minimize nesting, and return early.

```suggestion
        if (!isQuerySuitable) {
          return supplementaryQuery;
        }
```

Reply:
Fixed in https://github.com/grafana/grafana/pull/62097/commits/6bc2b4c9925c56d2260168076329133753fd2981?diff=split&w=1